### DISCUSSION TITLE
How to make renovate parse a (kinda) different structure of package.json

### DISCUSSION THREAD

**Discussion by abruzzi** [renovatebot/renovate#9278] [Request Help] [answered]
Hi renovate 

I have an issue (not a common usage to be honest) about how to config `renovate` to detect dependencies defined in a template file other than `package.json`.

In `create-react-app` you can use template to generate a new project (https://create-react-app.dev/docs/custom-templates/), it's using `template.json` as a template and will be merged to the generated package.json at runtime.

I can tell `renovate` to check the file by saying this:

```json
{
  "extends": ["config:base"],
  "npm": {
    "fileMatch": ["(^|/)template.json$"]
  },
}
```

which is fine, but the structure in `template.json` I believe not quite align with the structure that renovate is expecting:

```json
{
    "package": {
      "dependencies": {
         /**/
      },
      "eslintConfig": {
        "extends": ["react-app", "react-app/jest"]
      }
    }
  }
```

note there is a `package` section, and then `dependencies` under it.

So I'd like to ask is there a way to let renovate be aware of this structure or should I raise a feature request instead?

Also I noted that renovate does support the old format like the one defined here (https://github.com/cypress-io/cra-template-cypress/blob/master/template.json) in cypress-io (as you can see there is no `package` section in the file, which is the old format).

**Comment by viceice** [comment, marked as answer]
I think your only way is to use a regex manager

https://docs.renovatebot.com/configuration-options/#regexmanagers

**Reply by abruzzi** [reply]
Hi @viceice 

Thanks for replying, and `regexManager` worked:

```
  "regexManagers": [
    {
      "fileMatch": ["^template.json$"],
      "matchStringsStrategy": "recursive",
      "matchStrings": [
        "\"dependencies\":\\s*\\{[^}]*}",
        "\"(?<depName>.*)\"[^\"]*:\\s*\"(?<currentValue>.*)\""
      ],
      "datasourceTemplate": "npm"
    }
  ],
```

Thank you!
